Sources/RTVIClientIOSGeminiLiveWebSocket/GeminiLiveWebSocketTransport.swift

```diff
@@ -118,14 +119,18 @@ public class GeminiLiveWebSocketTransport: Transport {
     public func updateMic(micId: RTVIClientIOS.MediaDeviceId) async throws {
         audioManager.preferredAudioDevice = .init(deviceID: micId.id)
         // Changing preferred audio device probably changed actual audio device in use
-        updateSelectedMicIfNeeded()
+        handleDeviceChangeIfNeeded()
     }
     
     public func updateCam(camId: RTVIClientIOS.MediaDeviceId) async throws {
         logOperationNotSupported(#function)
     }
     
     public func selectedMic() -> RTVIClientIOS.MediaDeviceInfo? {
+        audioManager.availableDevices.first { $0.deviceID == audioManager.preferredAudioDeviceIfAvailable?.deviceID }?.toRtvi()
+    }
+    
+    public func actualMicInUse() -> RTVIClientIOS.MediaDeviceInfo? {
```

Comment:
Should this be Optional ? Is there any case where we wouldn't have a mic in use ?

Reply:
I was thinking that prior to `initDevices()` it didn't make sense for this to have a value